express-notes-api/index.js

```diff
@@ -0,0 +1,89 @@
+const data = require('./data');
+const fs = require('fs');
+const express = require('express');
+const app = express();
+
+app.get('/api/notes', (req, res) => {
+  const notesArray = [];
+  for (const key in data.notes) {
+    notesArray.push(data.notes[key]);
+  }
+  res.json(notesArray);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const id = Number(req.params.id);
+  if (isNaN(id)) {
+    res.status(400).send({ error: 'id must be a positive integer' });
+  } else if (!data.notes[id]) {
+    res.status(404).send({ error: `can not find note with id ${id}` });
+  } else {
+    res.send(data.notes[id]);
+  }
+});
+
+app.use(express.json());
+
+app.post('/api/notes', (req, res) => {
+  if (!req.body.content) {
+    res.status(404).send({ error: 'content is a required field' });
+  } else if (req.body.content) {
+    data.notes[data.nextId] = req.body;
+    data.notes[data.nextId].id = data.nextId;
+    data.nextId++;
+    const stringData = JSON.stringify(data, null, 2);
+    fs.writeFile('./data.json', stringData, 'utf8', err => {
+      if (err) {
+        console.error(err);
+        res.status(500).send({ error: 'an unexpected error occured.' });
+      } else {
+        res.send(data.notes[data.nextId - 1]);
+      }
```

Comment:
A successful creation should result in a `201` status (created) instead of just a `200` (ok).